src/pages/fieldManagement/settings/DefaultFieldTable.jsx

```diff
@@ -114,22 +114,57 @@ export default function DefaultFieldTable({
             setFormSettings(getInitialFormState(siteSettings));
             onCloseEditor();
           }}
-          onSubmit={() => {
-            if (editField.id === 'region') {
-              putSiteSetting(
+          onSubmit={async () => {
+            if (editField?.id === 'region') {
+              const success = await putSiteSetting(
                 editField.backendPath,
                 formSettings.regions,
-              ).then(success => {
-                if (success) onCloseEditor();
-              });
+              );
+              if (success) onCloseEditor();
             }
-            if (editField.id === 'species') {
-              putSiteSetting(
+            if (editField?.id === 'species') {
+              const success = await putSiteSetting(
                 editField.backendPath,
                 formSettings.species,
-              ).then(success => {
-                if (success) onCloseEditor();
-              });
+              );
+              if (success) onCloseEditor();
+            }
+            if (editField?.id === 'relationship') {
+              const newSettings = reduce(
+                formSettings.relationships,
+                (memo, currentRelationships, key) => {
+                  const dedupedRelationships = currentRelationships.filter(
+                    (entry, index) => {
+                      return (
+                        currentRelationships.indexOf(entry) === index
+                      );
+                    },
+                  );
+                  if (
+                    currentRelationships.length >
+                    dedupedRelationships.length
+                  ) {
+                    alert(
```

Comment:
dunno how i feel about using browser alerts... this is okay for now but eventually i want to think more deeply about this

Reply:
I figured that you would not love this. I initially tried using setError(), but the subsequent putSiteSetting() call ends up being successful and dismissing the dialog, so you see the "error" for a millisecond.
I also tried adding another hasDuplicate boolean before dismissal and ended up not using that either for a reason that I can't remember.
I guess we could do a nested dialog, but that seemed crazy. I'd like to chop that into a separate task if that's what is desired.

Reply:
let's merge this as-is now. but could you create a little ticket to look back at this file and assign it to me?